web/src/app/users/UserEditForm.tsx

```diff
@@ -0,0 +1,91 @@
+import { TextField, FormControlLabel, Checkbox } from '@mui/material'
+import Grid from '@mui/material/Grid'
+import React from 'react'
+import { FormContainer, FormField } from '../forms'
+import { FieldError } from '../util/errutil'
+
+export type Value = {
+  oldPassword: string
+  newPassword: string
+  confirmNewPassword: string
+  isAdmin: boolean
+}
+
+export interface UserEditFormProps {
+  value: Value
+  errors: Array<FieldError>
+  admin: boolean
+  onChange: (newValue: Value) => void
+}
+
+function UserEditForm(props: UserEditFormProps): JSX.Element {
+  const { value, errors, admin, onChange } = props
+
+  return (
+    <FormContainer
+      value={value}
+      errors={errors}
+      onChange={(val: Value) => {
+        onChange(val)
+      }}
+    >
+      <Grid container spacing={2}>
+        {!admin && (
+          <Grid item xs={12}>
+            <FormField
+              InputProps={{
+                'data-cy': 'oldPassword',
+              }}
```

Comment:
you can probably test off of `label` or something so we don't have this artifact (and the ones below) in the future if/when we are migrating to playwright

(no worries if this was the best solution you found though, we have a lot of `'data-cy'` tags throughout the app)